core/frontend/src/EnvironmentDecorations.ts

```diff
@@ -86,11 +86,17 @@ export class EnvironmentDecorations {
 
   public decorate(context: DecorateContext): void {
     const env = this._environment;
+
+    let sky: RenderGraphic | undefined = undefined;
     if (env.displaySky && this._sky.params) {
-      const sky = IModelApp.renderSystem.createSkyBox(this._sky.params);
-      if (sky)
-        context.setSkyBox(sky);
+      sky = IModelApp.renderSystem.createSkyBox(this._sky.params);
     }
+    if (env.displayAtmosphere && env.atmosphere) { // Atmosphere can only be drawn if sky is also drawn
+      sky = IModelApp.renderSystem.createSkyBox(this.createSkyGradientParams());
```

Comment:
You have potentially already created a skybox in the clause above, which you now discard here. Change the order of your conditions to make sure you create at most one skybox?

Reply:
Streamlined that into an if-else to only ever make 1 skybox